### ISSUE TITLE
a naive question about crf.

### ISSUE THREAD

**Issue by cuixiongwen** [kazuto1011/deeplab-pytorch#15]
Thanks for your excellent code!
I am a novice at crf. I notice that crf is used during evaling but not during training. Could someone tell me why? Thanks for your answer!

**Comment by kazuto1011** [comment]
I think the FCN training is like a part of CRF optimization. The CRF post-processing in evaluation is actually decoupled optimization using logits from the fixed FCN (mean field inference with a fixed unary potential). I followed the DeepLab v2 way. See section 3.3 of the original paper.
